File: Unknownbank/app/src/main/java/com/kasatria/kasatriaunknownbank/CreditCardAlmostThereScreen.kt
package com.kasatria.kasatriaunknownbank

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

import com.kasatria.kasatriaunknownbank.ui.theme.PrimaryBlue
import com.kasatria.kasatriaunknownbank.ui.theme.TextPrimary
import com.kasatria.kasatriaunknownbank.ui.theme.White


@Composable
fun CreditCardAlmostThereScreen(
    onBack: () -> Unit,
    onUploadNow: () -> Unit
) {

    Box(
        modifier = Modifier.fillMaxSize()
    ) {

        /*
         * Decorative background
         */

        Image(
            painter = painterResource(
                id = R.drawable.full_background_light
            ),
            contentDescription = null,
            modifier = Modifier.fillMaxSize(),
            contentScale = ContentScale.Crop
        )


        Column(
            modifier = Modifier
                .fillMaxSize()
                .statusBarsPadding()
        ) {

            /*
             * =================================
             * HEADER
             * =================================
             */

            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(56.dp)
            ) {

                IconButton(
                    onClick = onBack,

                    modifier = Modifier
                        .padding(start = 8.dp)
                        .align(
                            Alignment.CenterStart
                        )
                ) {

                    Icon(
                        imageVector =
                            Icons.AutoMirrored.Filled.ArrowBack,

                        contentDescription = "Back",

                        tint = TextPrimary
                    )
                }


                Text(
                    text =
                        "Applying for Bank World Mastercard",

                    modifier = Modifier
                        .align(Alignment.Center)
                        .padding(
                            start = 55.dp,
                            end = 30.dp
                        ),

                    color = Color(0xFF333333),

                    fontSize = 13.sp,

                    fontWeight =
                        FontWeight.Bold,

                    maxLines = 1
                )
            }


            /*
             * =================================
             * INTRO
             * =================================
             */

            Text(
                text = "You are almost there.",

                modifier = Modifier.padding(
                    start = 20.dp,
                    end = 20.dp,
                    top = 20.dp
                ),

                color = Color(0xFF333333),

                fontSize = 15.sp,

                fontWeight =
                    FontWeight.Bold
            )


            Text(
                text =
                    "Upload your documents to complete your application",

                modifier = Modifier.padding(
                    start = 20.dp,
                    end = 20.dp,
                    top = 5.dp
                ),

                color = Color(0xFF44474D),

                fontSize = 14.sp,

                lineHeight = 22.sp
            )


            Spacer(
                modifier =
                    Modifier.height(35.dp)
            )


            /*
             * =================================
             * APPLICATION CARD
             * =================================
             */

            ApplicationReferenceCard(
                modifier = Modifier.padding(
                    horizontal = 20.dp
                )
            )


            Spacer(
                modifier =
                    Modifier.height(26.dp)
            )


            /*
             * =================================
             * NOTES
             * =================================
             */

            Text(
                text =
                    "*You may close this page if you have uploaded your documents.",

                modifier = Modifier.padding(
                    horizontal = 20.dp
                ),

                color = Color(0xFF44474D),

                fontSize = 14.sp,

                lineHeight = 20.sp
            )


            Spacer(
                modifier =
                    Modifier.height(20.dp)
            )


            Text(
                text =
                    "*If you do not have your supporting documents ready you may opt to upload them later before your application expires.",

                modifier = Modifier.padding(
                    horizontal = 20.dp
                ),

                color = Color(0xFF44474D),

                fontSize = 14.sp,

                lineHeight = 20.sp
            )
        }


        /*
         * =================================
         * UPLOAD NOW
         * =================================
         */

        Button(
            onClick = onUploadNow,

            modifier = Modifier
                .align(
                    Alignment.BottomCenter
                )
                .navigationBarsPadding()
                .padding(
                    start = 16.dp,
                    end = 16.dp,
                    bottom = 16.dp
                )
                .fillMaxWidth()
                .height(46.dp),

            shape =
                RoundedCornerShape(100.dp),

            colors =
                ButtonDefaults.buttonColors(
                    containerColor =
                        PrimaryBlue,

                    contentColor =
                        White
                )
        ) {

            Text(
                text = "Upload Now",

                fontSize = 15.sp,

                fontWeight =
                    FontWeight.Bold
            )
        }
    }
}


@Composable
private fun ApplicationReferenceCard(
    modifier: Modifier = Modifier
) {

    val shape =
        RoundedCornerShape(12.dp)

    Column(
        modifier = modifier
            .fillMaxWidth()
            .shadow(
                elevation = 3.dp,
                shape = shape
            )
            .background(
                color = White,
                shape = shape
            )
            .border(
                width = 1.dp,
                color = Color(0xFFEEEEEE),
                shape = shape
            )
            .padding(
                start = 17.dp,
                end = 17.dp,
                top = 17.dp,
                bottom = 27.dp
            )
    ) {

        /*
         * Reference Number
         */

        Row(
            modifier =
                Modifier.fillMaxWidth(),

            horizontalArrangement =
                Arrangement.SpaceBetween,

            verticalAlignment =
                Alignment.CenterVertically
        ) {

            Text(
                text = "Reference Number",

                color = Color.Black,

                fontSize = 14.sp
            )


            Text(
                text = "21092484852030",

                color = Color.Black,

                fontSize = 14.sp,

                fontWeight =
                    FontWeight.Bold
            )
        }


        Spacer(
            modifier =
                Modifier.height(11.dp)
        )


        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(1.dp)
                .background(
                    Color(0xFFF2F4F6)
                )
        )


        Spacer(
            modifier =
                Modifier.height(20.dp)
        )


        /*
         * Selected card image
         * Figma: 280 x 175
         */

        Image(
            painter = painterResource(
                id =
                    R.drawable.application_card_world
            ),

            contentDescription =
                "Bank World Mastercard",

            modifier = Modifier
                .width(280.dp)
                .height(175.dp)
                .align(
                    Alignment.CenterHorizontally
                ),

            contentScale =
                ContentScale.Crop
        )


        Spacer(
            modifier =
                Modifier.height(15.dp)
        )


        Text(
            text =
                "Bank World Mastercard",

            modifier =
                Modifier.fillMaxWidth(),

            color =
                Color(0xFF333333),

            fontSize = 15.sp,

            textAlign =
                TextAlign.Center
        )
    }
}

File: unknownbank_flutter/lib/screens/credit_card_almost_there_screen.dart
import 'package:flutter/material.dart';

import '../models/credit_card_product.dart';
import '../theme/app_colors.dart';

class CreditCardAlmostThereScreen
    extends StatelessWidget {
  final CreditCardProduct product;
  final String applicationReference;
  final VoidCallback onUploadNow;

  const CreditCardAlmostThereScreen({
    super.key,
    required this.product,
    required this.applicationReference,
    required this.onUploadNow,
  });

  @override
  Widget build(BuildContext context) {
    final image = product.name ==
            'Bank World Mastercard'
        ? 'application_card_world.png'
        : product.image;

    final isWorld =
        product.name == 'Bank World Mastercard';

    return Scaffold(
      body: Stack(
        children: [
          Positioned.fill(
            child: Image.asset(
              'assets/images/full_background_light.png',
              fit: BoxFit.cover,
            ),
          ),

          SafeArea(
            child: Column(
              crossAxisAlignment:
                  CrossAxisAlignment.start,
              children: [
                SizedBox(
                  height: 56,
                  child: Stack(
                    alignment: Alignment.center,
                    children: [
                      Align(
                        alignment:
                            Alignment.centerLeft,
                        child: Padding(
                          padding:
                              const EdgeInsets.only(
                            left: 8,
                          ),
                          child: IconButton(
                            onPressed: () {
                              Navigator.of(context)
                                  .pop();
                            },
                            icon: const Icon(
                              Icons.arrow_back,
                              color:
                                  AppColors.textPrimary,
                            ),
                          ),
                        ),
                      ),

                      Padding(
                        padding:
                            const EdgeInsets.only(
                          left: 55,
                          right: 30,
                        ),
                        child: Text(
                          'Applying for ${product.name}',
                          maxLines: 1,
                          overflow:
                              TextOverflow.ellipsis,
                          style: const TextStyle(
                            fontSize: 13,
                            fontWeight:
                                FontWeight.bold,
                            color:
                                Color(0xFF333333),
                          ),
                        ),
                      ),
                    ],
                  ),
                ),

                const Padding(
                  padding: EdgeInsets.fromLTRB(
                    20,
                    20,
                    20,
                    0,
                  ),
                  child: Text(
                    'You are almost there.',
                    style: TextStyle(
                      fontSize: 15,
                      fontWeight: FontWeight.bold,
                      color: Color(0xFF333333),
                    ),
                  ),
                ),

                const Padding(
                  padding: EdgeInsets.fromLTRB(
                    20,
                    5,
                    20,
                    0,
                  ),
                  child: Text(
                    'Upload your documents to complete your application',
                    style: TextStyle(
                      fontSize: 14,
                      height: 1.57,
                      color: Color(0xFF44474D),
                    ),
                  ),
                ),

                const SizedBox(height: 35),

                Padding(
                  padding: const EdgeInsets.symmetric(
                    horizontal: 20,
                  ),
                  child: Container(
                    width: double.infinity,
                    padding: const EdgeInsets.fromLTRB(
                      17,
                      17,
                      17,
                      27,
                    ),
                    decoration: BoxDecoration(
                      color: Colors.white,
                      borderRadius:
                          BorderRadius.circular(12),
                      border: Border.all(
                        color:
                            const Color(0xFFEEEEEE),
                      ),
                      boxShadow: const [
                        BoxShadow(
                          color: Color(0x22000000),
                          blurRadius: 6,
                          offset: Offset(0, 2),
                        ),
                      ],
                    ),
                    child: Column(
                      children: [
                        Row(
                          mainAxisAlignment:
                              MainAxisAlignment
                                  .spaceBetween,
                          children: [
                            const Text(
                              'Reference Number',
                              style: TextStyle(
                                fontSize: 14,
                                color: Colors.black,
                              ),
                            ),

                            Text(
                              applicationReference,
                              style: const TextStyle(
                                fontSize: 14,
                                fontWeight:
                                    FontWeight.bold,
                                color: Colors.black,
                              ),
                            ),
                          ],
                        ),

                        const SizedBox(height: 11),

                        const Divider(
                          height: 1,
                          color:
                              Color(0xFFF2F4F6),
                        ),

                        const SizedBox(height: 20),

                        SizedBox(
                          width: 280,
                          height: 175,
                          child: Image.asset(
                            'assets/images/$image',
                            fit: isWorld
                                ? BoxFit.cover
                                : BoxFit.contain,
                          ),
                        ),

                        const SizedBox(height: 15),

                        Text(
                          product.name,
                          textAlign:
                              TextAlign.center,
                          style: const TextStyle(
                            fontSize: 15,
                            color:
                                Color(0xFF333333),
                          ),
                        ),
                      ],
                    ),
                  ),
                ),

                const SizedBox(height: 26),

                const Padding(
                  padding: EdgeInsets.symmetric(
                    horizontal: 20,
                  ),
                  child: Text(
                    '*You may close this page if you have uploaded your documents.',
                    style: TextStyle(
                      fontSize: 14,
                      height: 1.43,
                      color: Color(0xFF44474D),
                    ),
                  ),
                ),

                const SizedBox(height: 20),

                const Padding(
                  padding: EdgeInsets.symmetric(
                    horizontal: 20,
                  ),
                  child: Text(
                    '*If you do not have your supporting documents ready you may opt to upload them later before your application expires.',
                    style: TextStyle(
                      fontSize: 14,
                      height: 1.43,
                      color: Color(0xFF44474D),
                    ),
                  ),
                ),
              ],
            ),
          ),

          Positioned(
            left: 16,
            right: 16,
            bottom: 16,
            child: SafeArea(
              top: false,
              child: SizedBox(
                height: 46,
                child: FilledButton(
                  onPressed: onUploadNow,
                  style: FilledButton.styleFrom(
                    backgroundColor:
                        AppColors.primary,
                    shape:
                        RoundedRectangleBorder(
                      borderRadius:
                          BorderRadius.circular(
                        100,
                      ),
                    ),
                  ),
                  child: const Text(
                    'Upload Now',
                    style: TextStyle(
                      fontSize: 15,
                      fontWeight:
                          FontWeight.bold,
                    ),
                  ),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}